classes/Workspace.php

```diff
@@ -42,23 +44,37 @@ class Workspace
      */
     private $paths;
 
+    /**
+     * @var Filesystem
+     */
+    private $filesystem;
+
     /**
      * @param string[] $paths
      */
-    public function __construct(Translator $translator, array $paths)
+    public function __construct(Translator $translator, Filesystem $filesystem, array $paths)
     {
         $this->translator = $translator;
+        $this->filesystem = $filesystem;
         $this->paths = $paths;
     }
 
+    /**
+     * @throws IOException
+     */
     public function createFolders(): void
     {
         foreach ($this->paths as $path) {
-            if (!file_exists($path) && !@mkdir($path)) {
-                throw new Exception($this->translator->trans('Unable to create directory %s', [$path]));
+            if (!$this->filesystem->exists($path)) {
+                try {
+                    $this->filesystem->mkdir($path);
+                } catch (IOException $e) {
+                    throw new IOException($this->translator->trans('Unable to create directory %s: %s', [$path, $e->getMessage()]));
+                }
```

Comment:
No need to check anymore if the directory already exists. If it does, `Filesystem` will ignore the creation.